crates/bevy_ptr/src/lib.rs

```diff
@@ -198,6 +208,23 @@ pub struct PtrMut<'a, A: IsAligned = Aligned>(NonNull<u8>, PhantomData<(&'a mut
 #[repr(transparent)]
 pub struct OwningPtr<'a, A: IsAligned = Aligned>(NonNull<u8>, PhantomData<(&'a mut u8, A)>);
 
+/// A Box-like pointer for moving a value to a new memory location without needing to pass by
+/// value.
+///
+/// Conceptually represents ownership of whatever data is being pointed to and will call its
+/// `Drop` impl upon being dropped. This pointer is _not_ responsible for freeing
+/// the memory pointed to by this pointer as it may be pointing to an element in a `Vec` or
+/// to a local in a function etc.
+///
+/// This type tries to act "borrow-like" like which means that:
+/// - Pointer should be considered exclusive and mutable. It cannot be cloned as this would lead
+///   to aliased mutability and potentially use after free bugs.
+/// - It must always points to a valid value of whatever the pointee type is.
+/// - The lifetime `'a` accurately represents how long the pointer is valid for.
+/// - It does not support pointer arithmetic in any way.
+/// - Must be sufficiently aligned for the pointee type.
+pub struct MovingPtr<'a, T, A: IsAligned = Aligned>(NonNull<T>, PhantomData<(&'a mut T, A)>);
```

Comment:
The usage / meaning of the `A` generic is not clear to me from reading the docs.